src/problems/sudoku.clj

```diff
@@ -0,0 +1,124 @@
+(ns problems.sudoku)
+;;OH BOY KIDS HERE WE GO
+(defn distinct?
+  "Returns true if no two of the arguments are ="
+  [& more]
+  (let [old (count more)
+        set (into #{} more)
+        new (count set)]
+    (= new old)))
+
+;;## validation top level ##
+(defn every-element-in-vec-distinct?
+ "Returns True if every element in every vector is unique or nil.
+  Returns False otherwise."
+  [vecs]
+ (every? identity (map #(apply distinct? %) vecs)))
+
+(def every-vector-valid? (memoize every-element-in-vec-distinct?))
```

Comment:
What is up with all the `memoize` (here and below)? Again, this seems like a totally false economy, complicating things trying to optimize for speed when the thing doesn't even work yet.